### PR TITLE
SNOW-683060 add test coverage for code-coverage batch-5

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #SNOW-683060

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

### CODE REVIEW

File: src/snowflake/snowpark/window.py [hunk 1]
```diff
@@ -246,25 +246,6 @@ def range_between(self, start: int, end: int) -> "WindowSpec":
             SpecifiedWindowFrame(RangeFrame(), boundary_start, boundary_end),
         )
 
-    def _convert_boundary_to_expr(
```
Comment (sfc-gh-aalam) [inline src/snowflake/snowpark/window.py:249]: This is a duplicate function which is never used.
https://github.com/snowflakedb/snowpark-python/blob/main/src/snowflake/snowpark/window.py#L28

File: tests/integ/test_table_function.py [hunk 2]
```diff
@@ -113,3 +151,30 @@ def test_call_table_function(session):
...
+
+def test_table_function_negative(session):
+    split_to_table = table_function("split_to_table")
+    df = session.create_dataframe(
+        [["Hello World", "p1", 1], ["Hello Python", "p2", 2]],
+        schema=["text", "partition", "seq"],
+    )
+    with pytest.raises(ValueError) as ex_info:
+        df.join_table_function(split_to_table("text", lit(" "), breaking_arg=1))
+    assert "A table function shouldn't have both args and named args" in str(ex_info)
```
Comment (sfc-gh-sfan) [inline tests/integ/test_table_function.py:173]: maybe we could do `match` in `pytest.raises`?

File: tests/integ/test_table_function.py [hunk 3]
```diff
@@ -113,3 +151,30 @@ def test_call_table_function(session):
...
+
+    with pytest.raises(ValueError) as ex_info:
+        df.join_table_function(split_to_table("text", lit(" ")), "breaking_arg")
+    assert (
```
Comment (sfc-gh-sfan) [inline tests/integ/test_table_function.py:177]: maybe we could do `match` in `pytest.raises`?

File: tests/unit/test_types.py [hunk 4]
```diff
@@ -191,13 +191,40 @@ def test_sf_datatype_names():
...
+
+
 def test_struct_field_name():
     column_identifier = ColumnIdentifier("identifier")
     assert StructField(column_identifier, IntegerType(), False).name == "identifier"
     assert (
         str(StructField(column_identifier, IntegerType(), False))
         == "StructField('identifier', IntegerType(), nullable=False)"
     )
+    assert (column_identifier == 7) is False
```
Comment (sfc-gh-sfan) [inline tests/unit/test_types.py:224]: nit: `column_identifier != 7`

BTW, what is this magic number 7?

File: tests/unit/test_udf.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        return_value="database.schema"
+    )
+    fake_session._run_query = mock.Mock(side_effect=ProgrammingError())
+    fake_session.udf = UDFRegistration(fake_session)
+    with pytest.raises(SnowparkSQLException) as ex_info:
+        udf(lambda: 1, session=fake_session, return_type=IntegerType(), packages=[])
+    assert ex_info.value.error_code == "1304"
+    cleanup_registration_patch.assert_called()
+
+    fake_session._run_query = mock.Mock(side_effect=BaseException())
```
Comment (sfc-gh-sfan) [inline tests/unit/test_udf.py:30]: Let's add some dummy error message to the exception and try to match in in `pytest.raises`? `BaseException` is a bit too broad for `pytest.raises`.

File: tests/unit/test_udtf.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    with pytest.raises(SnowparkSQLException) as ex_info:
+        @udtf(output_schema=["num"], session=fake_session)
+        class UDTFProgrammingErrorTester:
+            def process(self, n: int) -> Iterable[Tuple[int]]:
+                yield (n,)
+
+    assert ex_info.value.error_code == "1304"
+    cleanup_registration_patch.assert_called()
+
+    fake_session._run_query = mock.Mock(side_effect=BaseException())
```
Comment (sfc-gh-sfan) [inline tests/unit/test_udtf.py:35]: Let's add some dummy error message to the exception and try to match in in `pytest.raises`?

**Review by sfc-gh-sfan — COMMENTED** [review]
LGTM. Let's change the PR title given it has more changes the sp registration, or break this down into multiple PRs?

File: tests/unit/test_types.py (hunk 4 above)
Comment (sfc-gh-aalam) [reply tests/unit/test_types.py:224]: To make sure we test `__eq__` with type other than `str` or `ColumnIdentifier`. I will add a comment about this in test

File: src/snowflake/snowpark/window.py [hunk 7]
```diff
@@ -246,25 +246,6 @@ def range_between(self, start: int, end: int) -> "WindowSpec":
...
-
-        if end == 0:
-            boundary_end = CurrentRow()
-        elif end >= Window.UNBOUNDED_FOLLOWING:
-            boundary_end = UnboundedFollowing()
-        else:
-            boundary_end = Literal(end)
-
-        return boundary_start, boundary_end
-
```
Comment (sfc-gh-yixie) [inline src/snowflake/snowpark/window.py:267]: Yeah I also see this isn't used. A module function does the same thing.

**Review by sfc-gh-yixie — APPROVED** [review]

**Pull request merged**